core/src/main/java/org/mule/runtime/core/privileged/transaction/XaTransaction.java

```diff
@@ -48,6 +49,7 @@
 public class XaTransaction extends AbstractTransaction {
 
   private static final Logger LOGGER = getLogger(XaTransaction.class);
+  private final boolean debugLogEnabled = LOGGER.isDebugEnabled();
```

Comment:
don't do this, the log configuration may be changed in runtime and this would cause changes there no not be taken